controllers/leaderelection/agent_leaderelection.go

```diff
@@ -0,0 +1,181 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. 2021
+ */
+
+package leaderelection
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/instana/instana-agent-operator/controllers/leaderelection/coordination_api"
+	"github.com/procyon-projects/chrono"
+	coreV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	LeaderElectionTask          chrono.ScheduledTask
+	LeaderElectionTaskScheduler chrono.TaskScheduler
+	KubernetesLeaderResourceId  = "com.instana.plugin.kubernetes.leader"
+
+	coordinationApi = coordination_api.New()
+)
+
+type LeaderElector struct {
+	Ctx    context.Context
+	Client client.Client
+	Scheme *runtime.Scheme
+	Log    logr.Logger
+}
+
+func (l *LeaderElector) StartCoordination(agentNameSpace string) error {
+	l.Log = ctrl.Log.WithName("leaderelector").WithName("InstanaAgent")
+	LeaderElectionTaskScheduler = chrono.NewDefaultTaskScheduler()
+	var err error
+	LeaderElectionTask, err = LeaderElectionTaskScheduler.ScheduleWithFixedDelay(func(ctx context.Context) {
+		var activePods map[string]coreV1.Pod
+		var err error
+		if activePods, err = l.fetchPods(agentNameSpace); err != nil {
+			l.Log.Error(err, "Unable to fetch agent pods for doing election")
+			return
+		}
+		l.pollAgentsAndAssignLeaders(activePods)
+
+		time.Sleep(5 * time.Second)
+	}, 5*time.Second)
+	if err == nil {
+		l.Log.Info("Task has been scheduled successfully.")
```

Comment:
Perhaps logging the error (if any) is even more important than logging when succeeded?